state-chain/runtime/src/chainflip/bitcoin_block_processor.rs

```diff
@@ -72,24 +94,57 @@ impl Hook<(BlockNumber, BtcEvent), ()> for ExecuteEventHook {
 	Default,
 )]
 pub struct ApplyRulesHook {}
-impl Hook<(BlockNumber, BlockNumber, BlockData), Vec<(BlockNumber, BtcEvent)>> for ApplyRulesHook {
+impl
+	Hook<
+		(BlockNumber, BlockNumber, BlockData),
+		Vec<(BlockNumber, BtcEvent<DepositWitness<Bitcoin>>)>,
+	> for ApplyRulesHook
+{
 	fn run(
 		&mut self,
 		(block, age, block_data): (BlockNumber, BlockNumber, BlockData),
-	) -> Vec<(BlockNumber, BtcEvent)> {
+	) -> Vec<(BlockNumber, BtcEvent<DepositWitness<Bitcoin>>)> {
 		// Prewitness rule
 		if age == 0 {
 			return block_data
 				.iter()
 				.map(|deposit_witness| (block, BtcEvent::PreWitness(deposit_witness.clone())))
-				.collect::<Vec<(BlockNumber, BtcEvent)>>();
+				.collect::<Vec<_>>();
 		}
 		//Full witness rule
 		if age == BitcoinIngressEgress::witness_safety_margin().unwrap() {
 			return block_data
 				.iter()
 				.map(|deposit_witness| (block, BtcEvent::Witness(deposit_witness.clone())))
-				.collect::<Vec<(BlockNumber, BtcEvent)>>();
+				.collect::<Vec<_>>();
+		}
+		vec![]
+	}
+}
+
+impl
+	Hook<
+		(BlockNumber, BlockNumber, BlockDataVaultDeposit),
+		Vec<(BlockNumber, BtcEvent<VaultDepositWitness<Runtime, BitcoinInstance>>)>,
+	> for ApplyRulesHook
+{
+	fn run(
+		&mut self,
+		(block, age, block_data): (BlockNumber, BlockNumber, BlockDataVaultDeposit),
+	) -> Vec<(BlockNumber, BtcEvent<VaultDepositWitness<Runtime, BitcoinInstance>>)> {
+		// Prewitness rule
+		if age == 0 {
+			return block_data
+				.iter()
+				.map(|deposit_witness| (block, BtcEvent::PreWitness(deposit_witness.clone())))
+				.collect::<Vec<_>>();
+		}
+		//Full witness rule
+		if age == BitcoinIngressEgress::witness_safety_margin().unwrap() {
```

Comment:
Something like:
```suggestion
		if age == BitcoinIngressEgress::witness_safety_margin().unwrap_or(berghain::BITCOIN_SAFETY_MARGIN) {
```
No need to hard unwrap here

Reply:
same for the deposit witnessing above too

Reply:
cannot access that value, it creates circular dependencies.
Defaulted to 0 for now